build.sh

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash -e
+# =====================================================================
+# Build script for Freifunk Fulda firmware runningn on Jenkins CI
+#
+# Source: https://github.com/freifunk-fulda
+# Contact: fffd-noc@lists.open-mail.net
+# Web: https://fulda.freifunk.net
+#
+# Credits:
+#   - Freifunk Darmstadt for your great support
+# =====================================================================
+
+# Default make options
+MAKEOPTS="-j 4 V=s"
+
+# Help function used in error messages and -h option
+usage() {
+  echo ""
+  echo "Build script for Freifunk-Fulda gluon firmware."
+  echo ""
+  echo "-c: Build command, [ update | download | build | sign | upload ]"
+  echo "-b: String for git branch, e.g. development"
+  echo "-d: Enable bash debug output"
+  echo "-h: Show this help"
+  echo "-m: Optional setting for make options. Default is \"-j 4\""
+  echo "-n: String for build number, e.g. b48"
+  echo "-w: Path for workspace, e.g. current work directoy"
+}
+
+# Evaluate arguments for build script.
+while getopts b:c:dhm:n:w: flag; do
+  case ${flag} in
+    b)
+      GIT_BRANCH=${OPTARG};
+      ;;
+    c)
+      case ${OPTARG} in
+        update)
+        COMMAND=${OPTARG}
+        ;;
+        download)
+        COMMAND=${OPTARG}
+        ;;
+        build)
+        COMMAND=${OPTARG}
+        ;;
+        sign)
+        COMMAND=${OPTARG}
+        ;;
+        upload)
+        COMMAND=${OPTARG}
+        ;;
+        *)
+          echo "Error: Invalid build command set."
+          usage
+          exit 1
+          ;;
+      esac
+      ;;
+    d)
+      set -x
+      ;;
+    h)
+      usage
+      ;;
+    m)
+      MAKEOPTS=${OPTARG}
+      ;;
+    n)
+      BUILD_NUMBER=${OPTARG}
+      ;;
+    w)
+      WORKSPACE=${OPTARG};
+      ;;
+    *)
+      usage
+      exit;
+      ;;
+  esac
+done
+
+shift $(( OPTIND - 1 ));
+
+# Sanity checks for required arguments
+if [ -z "${GIT_BRANCH}" ]; then
+  echo "Error: Git branch with -b is not set."
+  usage
+  exit 1
+fi
+
+if [ -z "${BUILD_NUMBER}" ]; then
+  echo "Error: Build number with -n is not set."
```

Comment:
Laut site.mk wird im Moment snapshot-$DATUM verwendet. In diesem Fall wird auch keine Versionsnummer eingesetzt.